commons/com.b2international.index.tests/src/com/b2international/index/revision/SingleDocumentRevisionIndexTest.java

```diff
@@ -54,6 +54,22 @@ public void updateRevision() throws Exception {
 		indexRevision(MAIN, data);
 		assertEquals(data, getRevision(MAIN, RevisionData.class, STORAGE_KEY1));
 	}
+	
+	@Test
+	public void stageNewRevisionMultipleTimes() throws Exception {
+		final RevisionData data = new RevisionData(STORAGE_KEY1, "fieldA", "fieldB");
+		StagingArea staging = index().prepareCommit(MAIN);
+		staging.stageNew(data, false);
+		
+		staging.stageNew(data, true);
+		assertEquals(1, staging.getNumberOfStagedObjects());
+		assertEquals(data, staging.getNewObject(RevisionData.class, STORAGE_KEY1));
```

Comment:
Perform an assert here that the isCommit flag did not change because there were no changes, so the original is still uncommitable.